plugin/storage/es/factory.go

```diff
@@ -124,65 +133,41 @@ func (f *Factory) InitFromViper(v *viper.Viper, _ *zap.Logger) {
 // configureFromOptions configures factory from Options struct.
 func (f *Factory) configureFromOptions(o *Options) {
 	f.Options = o
-	f.primaryConfig = f.Options.GetPrimary()
-	f.archiveConfig = f.Options.Get(archiveNamespace)
+	f.config = f.Options.GetPrimary()
 }
 
 // Initialize implements storage.Factory.
 func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger) error {
 	f.metricsFactory = metricsFactory
 	f.logger = logger
 
-	primaryClient, err := f.newClientFn(f.primaryConfig, logger, metricsFactory)
+	primaryClient, err := f.newClientFn(f.config, logger, metricsFactory)
 	if err != nil {
 		return fmt.Errorf("failed to create primary Elasticsearch client: %w", err)
 	}
-	f.primaryClient.Store(&primaryClient)
+	f.client.Store(&primaryClient)
 
-	if f.primaryConfig.Authentication.BasicAuthentication.PasswordFilePath != "" {
-		primaryWatcher, err := fswatcher.New([]string{f.primaryConfig.Authentication.BasicAuthentication.PasswordFilePath}, f.onPrimaryPasswordChange, f.logger)
+	if f.config.Authentication.BasicAuthentication.PasswordFilePath != "" {
+		primaryWatcher, err := fswatcher.New([]string{f.config.Authentication.BasicAuthentication.PasswordFilePath}, f.onPrimaryPasswordChange, f.logger)
 		if err != nil {
 			return fmt.Errorf("failed to create watcher for primary ES client's password: %w", err)
 		}
 		f.watchers = append(f.watchers, primaryWatcher)
 	}
 
-	if f.archiveConfig.Enabled {
-		archiveClient, err := f.newClientFn(f.archiveConfig, logger, metricsFactory)
-		if err != nil {
-			return fmt.Errorf("failed to create archive Elasticsearch client: %w", err)
-		}
-		f.archiveClient.Store(&archiveClient)
-
-		if f.archiveConfig.Authentication.BasicAuthentication.PasswordFilePath != "" {
-			archiveWatcher, err := fswatcher.New([]string{f.archiveConfig.Authentication.BasicAuthentication.PasswordFilePath}, f.onArchivePasswordChange, f.logger)
-			if err != nil {
-				return fmt.Errorf("failed to create watcher for archive ES client's password: %w", err)
-			}
-			f.watchers = append(f.watchers, archiveWatcher)
-		}
-	}
-
 	return nil
 }
 
 func (f *Factory) getPrimaryClient() es.Client {
-	if c := f.primaryClient.Load(); c != nil {
-		return *c
-	}
-	return nil
-}
-
-func (f *Factory) getArchiveClient() es.Client {
-	if c := f.archiveClient.Load(); c != nil {
+	if c := f.client.Load(); c != nil {
```

Comment:
(not for this PR) I am curious if we still need this indirection. The primary reason it was added was to support TLS cert reload. But we don't do file-watcher-based reloads anymore, so nothing will ever trigger resetting of the client, and the cert rotation should be supported via OTEL's configtls indirection (as confirmed by the unit tests where we simulate rotation of certs). One this PR is merged let's open a good-first-issue ticket.

Reply:
oh, maybe we still use file-watcher for password file, then we'd keep this.